# Basic Programs:

# Python program to add two numbers

In [2]:
num1 = int(input("Enter the 1st number:"))
num2 = int(input("Enter the 2nd number:"))
addition = num1+num2
print(addition)

Enter the 1st number:1
Enter the 2nd number:2
3


In [62]:
num1,num2 = input("Enter two value:").split()
add = int(num1)+int(num2)
print(add)

Enter two value: 1 4
5


In [9]:
def addition(num1,num2):
    add = int(num1) + int(num2)
    return add
    
addition(3,4)

7

# Maximum of two numbers in Python

In [12]:
num1 = int(input("Enter the 1st number:"))
num2 = int(input("Enter the 2nd number:"))
if num1>num2:
    print(num1)
else:
    print(num2)


Enter the 1st number:-1
Enter the 2nd number:-4
-1


In [15]:
num1 = int(input("Enter the 1st number:"))
num2 = int(input("Enter the 2nd number:"))
print(max(num1,num2))

Enter the 1st number:22
Enter the 2nd number:33
33


# Python Program for factorial of a number

In [65]:
def factorial(number):
    a = number
    b=number
    while b > 1:
        c=b-1
        fact = a * c
        a = fact
        b=c
    return fact
print("factorial is",factorial(6))

factorial is 720


In [57]:
number = int(input("Enter number"))
b=number
while b > 1:
    c=b-1
    fact = number * c
    number = fact
    b=c
print("Factorial is",fact)



Enter number6
Factorial is 720


In [55]:
# Python 3 program to find
# factorial of given number
def factorial(n):

    # single line to find factorial
    return 1 if (n==1 or n==0) else n * factorial(n - 1);

# Driver Code
num = 5;
print("Factorial of",num,"is",factorial(num))

# This code is contributed by Smitha Dinesh Semwal


Factorial of 5 is 120


# Python Program for simple interest

Simple interest formula is given by:
Simple Interest = (P x T x R)/100
Where,

P is the principle amount,
T is the time and,
R is the rate,

In [15]:
P,T,R = input("enter pricipal,time period and interest").split()
print('The Simple Interest is',int(P)*int(T)*int(R)//100)

enter pricipal,time period and interest10000 5 5
The Simple Interest is 2500


In [14]:
# Python3 program to find simple interest
# for given principal amount, time and
# rate of interest.
  
def simple_interest(p,t,r):
    print('The principal is', p)
    print('The time period is', t)
    print('The rate of interest is',r)
      
    si = (p * t * r)/100
      
    print('The Simple Interest is', si)
    
      
# Driver code
simple_interest(8, 6, 8)

The principal is 8
The time period is 6
The rate of interest is 8
The Simple Interest is 3.84


# Python Program for compound interest

Formula to calculate compound interest annually is given by: 

A = P(1 + R/100) t 

Compound Interest = A – P 

Where, 
A is amount, 
P is principle amount, 
R is the rate and, 
T is the time span,

In [26]:
# we have to use pow() function.
# it takes two arguments.(value and it is power by)


def compound_interest(Principle,Rate,Timespan):
    Amount = Principle*(pow((1+Rate/100),Timespan))
    print(" The Compound Interest is",Amount-Principle)
compound_interest(10000,10.25,5)

 The Compound Interest is 6288.946267774416


# Python Program to check Armstrong Number

Input : 153

Output : Yes

153 is an Armstrong number.
1x1x1 + 5x5x5 + 3x33 = 153

Input : 120

Output : No

120 is not a Armstrong number.
1x1x1 + 2x2x2 + 0x0x0 = 9

Input : 1253

Output : No

1253 is not a Armstrong Number
1x1x1x1 + 2x2x2x2 + 5x5x5x5 + 3x3x3x3 = 723



# How to find out power without using pow() function

In [46]:
# Function to calculate x raised to 
# the power y
def power(x, y):
      
    if y == 0:
        return 1
    if y % 2 == 0:
        return power(x, y // 2) * power(x, y // 2)
          
    return x * power(x, y // 2) * power(x, y // 2)

In [45]:
a=input("Enter number:")
length=len(a)
sums=0
for i in a:
    b=pow(int(i),l) ##pow() function is used.
    sums=sums+b
print(sums)
if sums==int(a):
    print("yes")
else:
    print("no")

1634
1634
yes


# Python Program for Program to find area of a circle

Area of a circle can simply be evaluated using following formula.

Area = pi * r2

where r is radius of circle 

In [50]:
def findarea(r):
    PI=3.142 # we know this
    return PI*(r*r);
print("Area is %.6f" % findarea(5));

Area is 78.550000


# Python program to print all Prime numbers in an Interval

In [52]:
# Python program to print all 
# prime number in an interval
# number should be greater than 1
start = 11
end = 25
  
for i in range(start, end+1):
    if i>1:
        for j in range(2,i):
            if(i % j==0):
                break
        else:
            print(i)

11
13
17
19
23


# Python program to check whether a number is Prime or not

In [54]:
# Python program to check if
# given number is prime or not
  
num = 11
  
# If given number is greater than 1
if num > 1:
  
    # Iterate from 2 to n / 2
    for i in range(2, int(num/2)+1):
  
        # If num is divisible by any number between
        # 2 and n / 2, it is not prime
        if (num % i) == 0:
            print(num, "is not a prime number")
            break
    else:
        print(num, "is a prime number")
else:
    print(num, "is not a prime number")

11 is a prime number
